src/linear_algebra/projection.lean

```diff
@@ -167,6 +167,77 @@ lemma linear_proj_of_is_compl_idempotent (h : is_compl p q) (x : E) :
     linear_proj_of_is_compl p q h x :=
 linear_proj_of_is_compl_apply_left h _
 
+lemma exists_unique_add_of_is_compl (hc : is_compl p q) (x : E) :
+  ∃ (u : p) (v : q), (u.1 + v.1 = x ∧ ∀ (r : p) (s : q),
```

Comment:
I think coercion rather than `.1` might be the simp-normal form

Reply:
Also, if you bundle p and q into a product, you can use exists_unique without the weirdness it would otherwise give